io/driver/driver_parameter.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2019 IBM
+# Author: Manvanthara Puttashankar<manvanth@linux.vnet.ibm.com>
+
+"""
+This Script verfies driver module parameter.
+"""
+import time
+import netifaces
+from avocado import main
+from avocado.utils import process
+from avocado.utils import linux_modules, genio
+from avocado.utils import configure_network
+from avocado import Test
+
+
+class Moduleparameter(Test):
+
+    """
+    This Script verfies driver module parameter.
+    """
+
+    def setUp(self):
+        """
+        get parameters
+        """
+        self.module = self.params.get('module', default=None)
+        interfaces = netifaces.interfaces()
+        self.ifaces = self.params.get("interface")
+        if self.ifaces not in interfaces:
+            self.cancel("%s interface is not available" % self.ifaces)
+        self.ipaddr = self.params.get("host_ip", default="")
+        self.netmask = self.params.get("netmask", default="")
+        self.peer = self.params.get("peer_ip")
+        if not self.peer:
+            self.cancel("No peer provided")
+        self.param_name = self.params.get('module_param_name', default=None)
+        self.param_value = self.params.get('module_param_value', default=None)
+        self.sysfs_chk = self.params.get('sysfs_check_required', default=None)
+        if self.ifaces[0:2] == 'ib':
+            configure_network.set_ip(self.ipaddr, self.netmask, self.ifaces,
+                                     interface_type='Infiniband')
+        else:
+            configure_network.set_ip(self.ipaddr, self.netmask, self.ifaces,
+                                     interface_type='Ethernet')
+        self.load_unload_sleep_time = 30
+        self.error_modules = []
+        self.mod_list = []
+        self.uname = linux_modules.platform.uname()[2]
+
+    def built_in_module(self, module):
+        """
+        checking whether the given module is built_in module or not
+        """
+        path = "/lib/modules/%s/modules.builtin" % self.uname
+        for each in genio.read_all_lines(path):
+            out = process.getoutput(each.split('/')[-1])
+            if module == out.split('.'[0]):
+                return True
+            return False
+
+    def ping_check(self, options):
```

Comment:
Please use https://github.com/avocado-framework/avocado/blob/master/avocado/utils/configure_network.py#L108

Reply:
ok can be done!!

Reply:
Changes included in latest code!!